variables.subscription.tf

```diff
@@ -197,3 +197,55 @@ variable "wait_for_subscription_before_subscription_operations" {
 The duration to wait after vending a subscription before performing subscription operations.
 DESCRIPTION
 }
+
+variable "subscription_budgets" {
+  type = map(object({
+    amount            = number
+    time_grain        = optional(string, "Monthly")
+    time_period_start = string
+    time_period_end   = string
+    notifications = map(object({
+      enabled        = bool
+      operator       = string # EqualTo, GreaterThan, GreaterThanOrEqualTo
+      threshold      = number # 0-1000 percent
+      threshold_type = string # Actual, Forecasted
+      contact_emails = optional(list(string), [])
+      contact_roles  = optional(list(string), [])
+      contact_groups = optional(list(string), [])
+      locale         = optional(string, "en-us")
+    }))
+  }))
+  default     = {}
+  description = <<DESCRIPTION
+The budgets to create for the subscription using the AzApi provider.
```

Comment:
Is it possible to expand on these docs to provide field level explanations? Ideally it should be that case that someone consuming this module does not need to lookup the expected value formats elsewhere. You can link off to other docs for allowed values if there is a long list.